src/shogun/base/SGObject.h

```diff
@@ -476,6 +476,47 @@ class CSGObject
 			demangled_type<T>().c_str());
 	}
 
+	/** Typed array getter for an object class parameter of a Shogun base class
+	* type, identified by a name and an index
+	*
+	* @param name name of the parameter array
+	* @param index index of the element in the array
+	*/
+	template <class T,
+			  class X = typename std::enable_if<is_sg_base<T>::value>::type>
+	T* get(const std::string& name, index_t index)
```

Comment:
this code is an almost copy-paste of the
```
	template <class T,
		      class X = typename std::enable_if<is_sg_base<T>::value>::type>
	void add(const std::string& name, T* value)
```

would be nice to unify.... and note: std::list, std::deque....

Reply:
yep agree, would be cool to do that. This was mostly to get something working for Giovanni, but I can look into doing this more general as well